docs/HailExpressionLanguage.md

```diff
@@ -56,6 +56,8 @@ Several Hail commands provide the ability to perform a broad array of computatio
      - max: `arr.max` -- valid only for numeric arrays, returns the minimum value
      - find: `arr.find(v => expr)` -- Returns the first non-missing element of `arr` for which `expr` is true.  If no element satisfies the predicate, `find` returns NA.
      - map: `arr.map(v => expr)` -- Returns a new array produced by applying `expr` to each element
+     - flatMap: `arr.flatMap(v => expr)` -- valid only for `expr` of type Array. Returns a new array by mapping each element of `arr` and concatenating the resulting arrays
```

Comment:
I don't think we can expect users to know whether `Array(Set(1, 2), Set(2, 3)).flatten` does .toArray on each set and then concatenates or takes the union of the sets and then does toArray...or even whether it returns a set or an array.  I can explain this in the docs, but we may reduce confusion by forcing users to be explicit in the mixed case, e.g. `arr.map(_.toArray).flatten` or `arr.toSet.flatten().toArray`. Thoughts?